scheduler/scheduler.go

```diff
@@ -149,9 +150,20 @@ func BuildDependencyGraph(c client.Interface, sel labels.Selector) (*DependencyG
 			depGraph[d.Parent].RequiredBy, depGraph[d.Child])
 	}
 
-	return &depGraph, nil
+	cycles := DetectCycles(depGraph)
```

Comment:
Thanks! Having the test boilerplate really helped. I will move the check to main func.